File: fix_arabic.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import sys

file_path = r'c:\Users\abdel\Desktop\تهنئة العيد\src\pages\EditorPage.jsx'

print('Reading file...')
with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

print(f'Total lines: {len(lines)}\n')

# Define the fixes (1-indexed line numbers and their correct content)
fixes = [
    (4045, '            <h3 style={{ margin: 0, fontSize: 16, fontWeight: 700 }}>التنسيق</h3>\n', 'التنسيق'),
    (4046, '            <p style={{ margin: 0, fontSize: 12, color: \'#888\' }}>الخط والألوان والتأثيرات</p>\n', 'الخط والألوان والتأثيرات'),
    (4052, None, 'الخط'),  # Check only
    (4069, None, 'ألوان النصوص'),  # Check only
    (4071, None, 'الرئيسي'),  # Check only
    (4072, None, 'الفرعي'),  # Check only
    (4073, None, 'المستلم'),  # Check only
    (4074, None, 'المرسل'),  # Check only
    (4080, None, 'التأثيرات'),  # Check only
    (4084, None, 'ظل النص'),  # Check only
    (4095, None, 'حدود النص'),  # Check only
    (4107, None, 'سمك الحدود'),  # Check only
    (4111, None, 'اللون'),  # Check only
    (4118, None, 'طبقة لونية'),  # Check only
]

modified = False
changed_lines = []

# Apply fixes
for line_num, correct_content, correct_arabic in fixes:
    idx = line_num - 1  # Convert to 0-indexed
    current_line = lines[idx]
    
    if correct_content is None:
        # Check only
        if correct_arabic in current_line:
            print(f'✓ Line {line_num}: Already has correct text "{correct_arabic}"')
        else:
            print(f'✗ Line {line_num}: Missing correct text')
            print(f'  Current: {current_line[:80]}')
            print(f'  Expected: "{correct_arabic}"\n')
    else:
        # Replace the line
        if current_line != correct_content:
            print(f'Fixing line {line_num}...')
            lines[idx] = correct_content
            modified = True
            changed_lines.append(line_num)
        else:
            print(f'✓ Line {line_num}: Already correct')

if modified:
    print(f'\n✓ Modified {len(changed_lines)} lines: {", ".join(map(str, changed_lines))}')
    print('Writing file back as UTF-8...')
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.writelines(lines)
    print('✓ File written successfully\n')
else:
    print('\n✓ No modifications needed\n')

# Verify by reading lines 4038-4130
print('=== VERIFICATION: Lines 4038-4130 ===\n')
verify_start = 4037  # 0-indexed
verify_end = 4130

target_lines = [4045, 4046, 4052, 4069, 4071, 4072, 4073, 4074, 4080, 4084, 4095, 4107, 4111, 4118]

for i in range(verify_start, min(verify_end, len(lines))):
    line_num = i + 1
    if line_num in target_lines:
        print(f'>>> Line {line_num}:')
        print(f'    {lines[i]}', end='')

print('\n✓ Script completed')

File: fix_arabic.js
const fs = require('fs');
const path = require('path');

const filePath = 'c:\\Users\\abdel\\Desktop\\تهنئة العيد\\src\\pages\\EditorPage.jsx';

// Read file as UTF-8
console.log('Reading file...');
const content = fs.readFileSync(filePath, 'utf8');
const lines = content.split('\n');

console.log(`Total lines: ${lines.length}\n`);

// Define the fixes with line numbers and correct content
const fixes = [
  {
    lineNum: 4045,
    correctContent: '            <h3 style={{ margin: 0, fontSize: 16, fontWeight: 700 }}>التنسيق</h3>',
    correctArabic: 'التنسيق'
  },
  {
    lineNum: 4046,
    correctContent: '            <p style={{ margin: 0, fontSize: 12, color: \'#888\' }}>الخط والألوان والتأثيرات</p>',
    correctArabic: 'الخط والألوان والتأثيرات'
  },
  {
    lineNum: 4052,
    correctArabic: 'الخط',
    checkOnly: true
  },
  {
    lineNum: 4069,
    correctArabic: 'ألوان النصوص',
    checkOnly: true
  },
  {
    lineNum: 4071,
    correctArabic: 'الرئيسي',
    checkOnly: true
  },
  {
    lineNum: 4072,
    correctArabic: 'الفرعي',
    checkOnly: true
  },
  {
    lineNum: 4073,
    correctArabic: 'المستلم',
    checkOnly: true
  },
  {
    lineNum: 4074,
    correctArabic: 'المرسل',
    checkOnly: true
  },
  {
    lineNum: 4080,
    correctArabic: 'التأثيرات',
    checkOnly: true
  },
  {
    lineNum: 4084,
    correctArabic: 'ظل النص',
    checkOnly: true
  },
  {
    lineNum: 4095,
    correctArabic: 'حدود النص',
    checkOnly: true
  },
  {
    lineNum: 4107,
    correctArabic: 'سمك الحدود',
    checkOnly: true
  },
  {
    lineNum: 4111,
    correctArabic: 'اللون',
    checkOnly: true
  },
  {
    lineNum: 4118,
    correctArabic: 'طبقة لونية',
    checkOnly: true
  }
];

let modified = false;
const changedLines = [];

// Apply fixes
for (const fix of fixes) {
  const idx = fix.lineNum - 1; // Convert to 0-indexed
  const currentLine = lines[idx];
  
  if (fix.checkOnly) {
    // Just check if the correct text exists
    if (currentLine.includes(fix.correctArabic)) {
      console.log(`✓ Line ${fix.lineNum}: Already has correct text "${fix.correctArabic}"`);
    } else {
      console.log(`✗ Line ${fix.lineNum}: Missing correct text`);
      console.log(`  Current: ${currentLine.substring(0, 80)}`);
      console.log(`  Expected: "${fix.correctArabic}"\n`);
    }
  } else {
    // Replace the full line
    if (currentLine !== fix.correctContent) {
      console.log(`Fixing line ${fix.lineNum}...`);
      lines[idx] = fix.correctContent;
      modified = true;
      changedLines.push(fix.lineNum);
    } else {
      console.log(`✓ Line ${fix.lineNum}: Already correct`);
    }
  }
}

if (modified) {
  console.log(`\n✓ Modified ${changedLines.length} lines: ${changedLines.join(', ')}`);
  console.log('Writing file back as UTF-8...');
  
  const newContent = lines.join('\n');
  fs.writeFileSync(filePath, newContent, { encoding: 'utf8' });
  console.log('✓ File written successfully\n');
} else {
  console.log('\n✓ No modifications needed\n');
}

// Verify by reading lines 4038-4130
console.log('=== VERIFICATION: Lines 4038-4130 ===\n');
const verifyStart = 4037; // 0-indexed
const verifyEnd = 4130;

for (let i = verifyStart; i < verifyEnd && i < lines.length; i++) {
  const lineNum = i + 1;
  // Highlight the target lines
  const isTarget = [4045, 4046, 4052, 4069, 4071, 4072, 4073, 4074, 4080, 4084, 4095, 4107, 4111, 4118].includes(lineNum);
  if (isTarget) {
    console.log(`>>> Line ${lineNum}:`);
    console.log(`    ${lines[i]}`);
  }
}

console.log('\n✓ Script completed');
